File: src/main/java/io/junseok/mealmate/domain/restaurant/repository/RestaurantRepository.java
package io.junseok.mealmate.domain.restaurant.repository;

import io.junseok.mealmate.domain.restaurant.entity.Restaurant;
import java.util.List;
import org.springframework.data.jpa.repository.JpaRepository;

public interface RestaurantRepository extends JpaRepository<Restaurant,Long> {
    boolean existsByRestaurantName(String restaurantName);

    List<Restaurant> findAllByRestaurantType(String restaurantType);

    List<Restaurant> find3TopByLikeCount();
}

File: src/main/java/io/junseok/mealmate/domain/restaurant/repository/RestaurantRepository.kt
package io.junseok.mealmate.domain.restaurant.repository

import io.junseok.mealmate.domain.restaurant.entity.Restaurant
import org.springframework.data.jpa.repository.JpaRepository

interface RestaurantRepository : JpaRepository<Restaurant, Long> {
    fun existsByRestaurantName(restaurantName: String): Boolean

    fun findAllByRestaurantType(restaurantType: String): List<Restaurant>?

    fun findTop3ByOrderByLikeCountDesc(): List<Restaurant>
}
